http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitorInterp.scala

```diff
@@ -0,0 +1,690 @@
+                  }
+                case (false, true) =>
+                  failedRuntime(monitorContext,
+                    runtimeAndRuntimeConfig,
+                    RuntimeErrorDetails(-1, Some(s"user script ${userScript.map(_.toUri).getOrElse("")} failed")))
+                case (true, false) =>
+                  failedRuntime(monitorContext,
+                    runtimeAndRuntimeConfig,
+                    RuntimeErrorDetails(-1, Some(s"user startUp script ${userStartupScript.map(_.toUri).getOrElse("")} failed")))
+                case (false, false) =>
+                  failedRuntime(monitorContext,
+                    runtimeAndRuntimeConfig,
+                    RuntimeErrorDetails(
+                      -1,
+                      Some(s"both user script ${userScript.map(_.toUri).getOrElse("")} and startUp ${userStartupScript.map(_.toUri).getOrElse("")} script failed")
+                    ))
+              }
+            } yield r
+          case ss =>
+            failedRuntime(
+              monitorContext,
+              runtimeAndRuntimeConfig,
+              RuntimeErrorDetails(-1, Some(s"unexpected GCE instance status ${ss} when trying to creating an instance"))
+            )
+        }
+      } yield r
+  }
+
+  private def startingRuntime(
+    instance: Option[Instance],
+    monitorContext: MonitorContext,
+    runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig,
+  )(implicit ev: ApplicativeAsk[F, AppContext]): F[CheckResult] = instance match {
+    case None =>
+      logger
+        .error(
+          s"${monitorContext} | Fail retrieve instance status when trying to start runtime ${runtimeAndRuntimeConfig.runtime.projectNameString}"
+        )
+        .as(((), None))
+    case Some(i) =>
+      for {
+        gceStatus <- F.fromEither(GceInstanceStatus.withNameInsensitiveOption(i.getStatus).toRight(new Exception(s"Unknown GCE instance status ${i.getStatus}")))
+        startableStatuses = Set(
+          GceInstanceStatus.Suspending,
+          GceInstanceStatus.Provisioning,
+          GceInstanceStatus.Staging,
+          GceInstanceStatus.Terminated,
+          GceInstanceStatus.Suspended,
+          GceInstanceStatus.Stopping,
+        )
+        r <- gceStatus match {
+          case s if(startableStatuses.contains(s)) =>
+            checkAgain(monitorContext, runtimeAndRuntimeConfig, Some(s"Instance is still in ${gceStatus}"))
+          case GceInstanceStatus.Running =>
+            val userStartupScript = getUserScript(i)
+
+            for {
+              startUpScriptSuccess <- checkUserScripts(userStartupScript)
+              r <- if (startUpScriptSuccess)
+                getInstanceIP(i) match {
+                  case Some(ip) =>
+                    // It takes a bit for jupyter to startup, hence wait 5 seconds before we check jupyter
+                    Timer[F].sleep(8 seconds) >> handleCheckTools(monitorContext, runtimeAndRuntimeConfig, ip, List.empty)
```

Comment:
Move to config

Reply:
these are largely experimental...they'll be moved to config before merge